test/integration/targets/dnf/tasks/nobest.yml

```diff
@@ -0,0 +1,24 @@
+- name: Install dnf-plugins-core in order to use dnf config-manager
+  dnf:
+    name: dnf-plugins-core
+    state: present
+
+- name: Add docker-ce repo (Only RedHat & CentOS)
+  shell: dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo
+  when: (ansible_distribution in ['RedHat', 'CentOS'])
+
+- name: Add docker-ce repo (Only Fedora)
+  shell: dnf config-manager --add-repo=https://download.docker.com/linux/fedora/docker-ce.repo
+  when: (ansible_distribution in ['Fedora'])
```

Comment:
The test should clean up any changes it makes to the running container to avoid causing problems for other tests run on this same image.

Reply:
I've added a task uninstalling `docker-ce` at last, as successfully removing package is tested on another tests (`dnf.yml`), I didn't add an assert task verifying uninstalling of `docker-ce` package, is it necessary?

Reply:
Also `dnf config-manager` doesn't have something like `remove-repo`, therefore I should remove it manually from `/etc/yum.repos.d/docker-ce.repo`, is this OK?

Reply:
Sorry to bother you multiple times with multiple comments, shall I remove dnf-plugins-core package too?

Reply:
FWIW, I don't have any opinion on `dnf-plugins-core` removal because it should never cause an issues with other tests since it's installed on basically every default install of Fedora, CentOS, or RHEL (8+ for the latter two) these days but it's probably best for test cleanliness that any change inflicted upon the test environment be undone for sanity.

:+1:

Reply:
> I didn't add an assert task verifying uninstalling of docker-ce package, is it necessary?

No, it's not necessary since we're not testing that removal works (which is covered by other tests). We just want the test to clean up after itself.

> Also dnf config-manager doesn't have something like --remove-repo, therefore I should have removed it manually from /etc/yum.repos.d/docker-ce.repo, is this OK?

Yes

> Sorry to bother you multiple times with multiple comments, shall I remove dnf-plugins-core package too?

It's no bother at all. I agree with @maxamillion: it's probably ok to leave that package. I was mainly concerned with the Docker packages and repositories since those may interfere with other tests.

For most PRs, we only run a small subset of tests. Each night we run _all_ the tests, so problematic interaction between tests only shows up in the nightly runs. Practicing good "test hygiene" is a good way to avoid issues with the nightly test runs.